### DISCUSSION TITLE
Move away from the verb 'Toggle' in the UI

### DISCUSSION THREAD

**Discussion by afercia** [WordPress/gutenberg#42492] [General]
Previously on core Trac:

**Remove use of "Toggle" in strings**
https://core.trac.wordpress.org/ticket/34753

I'd like to propose to move away from the verb 'Toggle' used for various user interface controls because of a few main reasons and I'd like to hear from the contributors community (including translators).

- It's used inconsistently.
- It makes strings unnecessary long.
- It's difficult to translate properly in many languages.

**Inconsistent**
Some buttons in the UI toggle the visibility of panels/sidebars/popovers and the like. Some of these button names use the verb 'Toggle', for example:
- Toggle navigation
- Toggle block inserter

<img width="800" alt="Toggle navigation Toggle block inserter" src="https://user-images.githubusercontent.com/1682452/179459809-a961a2a8-14ae-415f-9460-3ea36f7d82d5.png">

Some buttons do the same thing: they toggle the visibility of panels/sidebars/popovers but they do _not_ use the verb 'Toggle'. For example:
- List View
- Settings

<img width="807" alt="List View Settings" src="https://user-images.githubusercontent.com/1682452/179460422-e32a4d81-ee43-490c-8e21-9b68bbb4bcdf.png">

By default, most of these button names appear only in the button tooltips but it's important to remind these names are always exposed to assistive technology users. They would benefit from more consistent, shorter, names.

**Difficult to translate**
In many languages there are equivalent verbs but in most of the cases either they don't fully translate the original meaning or they're very, very, technical terms and add cognitive load for users. For example in my language (Italian) 'toggle' can be translated to:
- alternare: doesn't fully communicate the original meaning, it's more 'to alternate'
- commutare: it's an very technical term

In most of the cases, WordPress translators are forced to use _two_ verbs to communicate the two different actions associated with a button, which makes the strings even longer. A few example in a couple languages:

**Italian**
- Attiva/disattiva la navigazione
- Attiva/disattiva l'inserter dei blocchi

<img width="800" alt="Toggle navigation it" src="https://user-images.githubusercontent.com/1682452/179463658-8bb0d794-963e-4196-9ecd-4f34b46ac276.png">

**French:**
- Permuter la navigation
- Ouvrir/fermer l'outil d'insertion de blocs

<img width="800" alt="Toggle navigation fr" src="https://user-images.githubusercontent.com/1682452/179464182-34ff4196-6561-4dd2-a54e-baa04432035a.png">

In this case, the two 'toggle' verbs are translated inconsistently which is one more signal of the troubles translators have to face with this verb. 

Worth also mentioning there's no consistent pattern with the naming of some controls. Should control names use a verb and communicate the control action or should they communicate the associated feature name? Also, what is the actual associated action? Is it toggling the visibility (e.g. 'Toggle navigation') or maybe the action users can take with that tool (e.g. 'Navigate')? That's for a separate discussion, I guess :) 

**Your screenshots**
Please do feel free to add screenshots of more 'Toggle' verbs translated in other languages. The more examples, the better.

Any thoughts and contributions are very welcome.

**Comment by aristath** [comment]
In Greek the word `toggle` is so difficult to translate that it usually results in long debates about how to translate it in each instance separately. There is no word for "Toggle", so it's always a rough approximation that usually end up being too long, or completely meaningless. Since there is no word for toggle, other languages need to be more verbose.

Right now these strings are not even translated into Greek because of the difficulties mentioned above... The rest of the UI is in Greek but those strings remain in English 😬

**Comment by SergeyBiryukov** [comment]
In Russian, "toggle" can be translated as an equivalent of "switch", but that does not always make sense in context, so it is also often translated as open/close, show/hide, turn on/off, etc.

![GB42492](https://user-images.githubusercontent.com/144512/179632398-856a542a-cff3-40ff-9fd3-45964a5930fd.png)

Looking at the screenshots here, this seems OK, but makes the labels a bit too long, perhaps longer than intended.

**Comment by carolinan** [comment]
In the Swedish translation, we have to use "Turn on or off" for "toggle":

![Screen Shot 2022-08-04 at 08 32 32](https://user-images.githubusercontent.com/7422055/182779119-b9d0100f-3d15-4351-974f-2c5c66c91884.png)

**Comment by afercia** [comment]
As of May 2024, I still see at least 12 occurrences of the verb 'toggle' in translatable strings:
- `label = __( 'Toggle full height' ),`
- `helpText = __( 'Toggle to show a large initial letter.' );`
- `title={ __( 'Toggle search label' ) }`
- `__( 'Double tap to toggle setting' )`
- `description: __( 'Toggle fullscreen mode.' ),`
- `label: __( 'Toggle settings sidebar' ),`
- `label: __( 'Toggle block inspector' ),`
- `label: __( 'Toggle settings sidebar' ),`
- `label: __( 'Toggle block inspector' ),`
- `label: __( 'Toggle spotlight' ),`
- `label: __( 'Toggle top toolbar' ),`
- `description: __( 'Toggle distraction free mode.' ),`
